crates/gitbutler-hunk-dependency/src/builder.rs

```diff
@@ -0,0 +1,128 @@
+use std::{
+    collections::HashMap,
+    path::{Path, PathBuf},
+};
+
+use gitbutler_stack::StackId;
+
+use crate::{diff::Diff, hunk::DependencyHunk, stack::DependencyStack};
+
+/// Calculates dependencies between workspace changes and workspace commits.
+///
+/// What we ultimately want to understand is, given an uncommitted change
+/// in some file, do the old line numbers intersect with any commmit(s) in
+/// the workspace?
+///
+/// The problem we have to overcome is that we the workspace changes are
+/// produced by diffing the working directory against the workspace commit.
+/// It means changes from one stack can offset line numbers in changes from
+/// a different stack. The most intuitive way of checking if they touch
+/// the same lines is to use regular git blame, but it suffers from two
+/// problems, 1) speed and 2) lack of --reverse flag in git2. The latter
+/// means we can't detect intersections with deleted lines.
+///
+/// If we don't calculate these dependencies correctly it means a user
+/// might be able to move a hunk into a stack where it cannot be committed.
+///
+/// So the solution here is that we build up the same information we would
+/// get from blame by adding diffs together.
+#[derive(Debug, Default, PartialEq, Clone)]
+pub struct HunkDependencyBuilder {
+    stacks: HashMap<StackId, DependencyStack>,
+}
+
+impl HunkDependencyBuilder {
```

Comment:
Is it a builder really? A builder would typically have methods that return the builder itself, so that they can be stringed together and then a build() that produces the thing that it builds

Reply:
You're right. I wonder what could be a better name.. 🤔